File: app/src/main/java/com/tanacom/introrealm/java/CountryDialog.java
package com.tanacom.introrealm.java;


import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import com.tanacom.introrealm.R;

import java.util.Objects;

public class CountryDialog extends DialogFragment {

    CountryListener listerner;

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        final AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());

        final LayoutInflater inflater = getActivity().getLayoutInflater();

        final View dialogLay = inflater.inflate(R.layout.country_dialog, null, true);

        final EditText editText = dialogLay.findViewById(R.id.et_country);

        builder.setView(dialogLay)
                .setPositiveButton(
                        "Add Country", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialogInterface, int i) {

                                String country = editText.getText().toString();

                                listerner.onAddCountry(country);

                            }
                        })
                .setNegativeButton("Cancel", new DialogInterface.OnClickListener() {

                            @Override
                            public void onClick(DialogInterface dialogInterface, int i) {
                                Objects.requireNonNull(CountryDialog.this.getDialog()).cancel();
                                listerner.onCancel(CountryDialog.this);
                            }

                        }

                );

        return builder.create();
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);

        try {
            listerner = (CountryListener) context;
        } catch (ClassCastException e) {

            throw new ClassCastException(getActivity().toString() + "must implement interface");
        }


    }


}

File: app/src/main/java/com/tanacom/introrealm/kotlin/CountryDialog.kt
package com.tanacom.introrealm.kotlin

import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.widget.EditText
import androidx.fragment.app.DialogFragment
import com.tanacom.introrealm.R
import com.tanacom.introrealm.kotlin.Constants.ADD_COUNTRY
import com.tanacom.introrealm.kotlin.Constants.CANCEL
import java.util.*

class CountryDialog : DialogFragment() {

    var listerner: CountryListener? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val builder = AlertDialog.Builder(activity)
        val inflater = activity!!.layoutInflater
        val dialogLay = inflater.inflate(R.layout.country_dialog, null, true)
        val editText = dialogLay.findViewById<EditText>(R.id.et_country)
        builder.setView(dialogLay)
            .setPositiveButton(
                ADD_COUNTRY
            ) { _, i ->
                val country = editText.text.toString()
                listerner!!.onAddCountry(country)
            }

//            .setNeutralButton("Sync") { _, _ ->
//                listerner!!.onSync()
//            }

            .setNegativeButton(
                CANCEL
            ) { _, i ->
                Objects.requireNonNull(this@CountryDialog.dialog)!!.cancel()
                listerner!!.onCancel(this@CountryDialog)
            }
        return builder.create()
    }

    override fun onAttach(context: Context) {

        super.onAttach(context)
        listerner = try {
            context as CountryListener
        } catch (e: ClassCastException) {
            throw ClassCastException(activity.toString() + "must implement interface")
        }
    }


}